definitions/marts/looker_studio_marts/ecf/ls_ecf_assurance_bad_shape_1_out_of_sequence.sqlx

```diff
@@ -0,0 +1,163 @@
+  /* Only declarations eligible for funding are permitted for the calculation*/
+  extended_declarations AS (
+  SELECT
+    declaration_id,
+    declaration_type,
+    declaration_created_at_date,
+    declaration_date,
+    participant_profile_id,
+    state,
+    cpd_lead_provider_name,
+    statement_id,
+    schedule_identifier,
+    participant_type,
+    cohort,
+    statement_cohort,
+    trn,
+    declaration_sequence_list.expected_declaration_sequence_position,
+    /* This date logic in the 'order by' exists to ensure that if a declaration is out of order compared to declarations received earlier on a different date but also evidenced on the same day as a declaration that should be later in the sequence, that declaration won't eclipse the prior declaration and prevent the qualify statement from working at the end of this mart*/
+    /* For example, if you were to evidence a retained 3 declaration on Sep 15th and then on Sep 17th evidence a completed declaration at 9AM and then a retained 2 declaration at 10 AM,for the purposes of the queries below, you would want the retained 2 declaration ordered immediately after the retained 3 declaration and then the completed declaration ordered after the retained 2 declaration - So Ret 3, Ret 2, Completed */
+    /* This prevents this misordered retained 2 declaration from being excluded by the qualify statement excluding declarations out of order simply because they were evidenced on the same date, as this ordering logic compares it to the retained 3 declaration instead of the completed declaration. */
+    row_number() over (partition by participant_profile_id order by date(declaration_date) asc, expected_declaration_sequence_position asc) as actual_evidenced_sequence_position,
+  FROM
+    ${ref('ecf_declarations')} declarations
+  INNER JOIN
+    declaration_sequence_list
+  USING
+    (declaration_type)
+  WHERE
+    DATE(declaration_created_at_date) BETWEEN declaration_date_start
+    AND declaration_date_end
+    and cohort >= 2023
+    AND state IN ('eligible',
+      'payable',
+      'paid'))
+      /* This identifies the declarations evidenced before and after a given record's declaration based on declaration_date. It checks if the permitted sequence number of the record's declaration falls out of order compared to the maximum expected order of the declarations evidenced before or compared to the minimum expected order of the declarations evidenced after*/
+      /* It does not consider the declaration to fall under this bad shape if the declaration is evidenced on the same date as the prior or next declarations, this is covered under another bad shape */
+      /* It does not consider the declaration to fall under this bad shape if this participant is on a 'reduced' schedule and the declaration is a 'completed' type */
+SELECT
+  *,
+   MAX(expected_declaration_sequence_position) OVER (

```

Comment:
Suggest using a WINDOW block to reduce duplication here and allow you to describe the windows more clearly

Reply:
Also consider using MAX_BY and MIN_BY in this block. https://cloud.google.com/bigquery/docs/reference/standard-sql/aggregate_functions#max_by